eclair-core/src/test/scala/fr/acinq/eclair/payment/receive/InvoicePurgerSpec.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.payment.receive
+
+import akka.actor.typed.scaladsl.adapter.ClassicActorSystemOps
+import akka.testkit.TestProbe
+import fr.acinq.bitcoin.Block
+import fr.acinq.eclair.TestDatabases.TestPgDatabases
+import fr.acinq.eclair.db.{IncomingPayment, IncomingPaymentStatus, PaymentType, PaymentsDbSpec}
+import fr.acinq.eclair.payment.Bolt11Invoice
+import fr.acinq.eclair.payment.receive.InvoicePurger.{PurgeCompleted, PurgeEvent}
+import fr.acinq.eclair.{CltvExpiryDelta, MilliSatoshiLong, TestKitBaseClass, TimestampMilli, TimestampMilliLong, TimestampSecond, TimestampSecondLong, randomBytes32}
+import org.scalatest.funsuite.AnyFunSuiteLike
+
+import scala.concurrent.duration.DurationInt
+
+class InvoicePurgerSpec extends TestKitBaseClass with AnyFunSuiteLike {
+
+  import PaymentsDbSpec._
+
+  test("purge invoices") {
+    val dbs = TestPgDatabases()
+    val db = dbs.db.payments
+    val count = 10
+
+    // create expired invoices
+    val expiredInvoices = Seq.fill(count)(Bolt11Invoice(Block.TestnetGenesisBlock.hash, Some(100 msat), randomBytes32(), alicePriv, Left("expired invoice"), CltvExpiryDelta(18),
+      timestamp = 1 unixsec))
+    val expiredPayments = expiredInvoices.map( invoice => IncomingPayment(invoice, randomBytes32(), PaymentType.Standard, invoice.createdAt.toTimestampMilli, IncomingPaymentStatus.Expired))
+    expiredInvoices.lazyZip(expiredPayments).foreach((invoice, payment) => db.addIncomingPayment(invoice, payment.paymentPreimage))
+
+    // create pending invoices
+    val pendingInvoices = Seq.fill(count)(Bolt11Invoice(Block.TestnetGenesisBlock.hash, Some(100 msat), randomBytes32(), alicePriv, Left("pending invoice"), CltvExpiryDelta(18),
+      timestamp = TimestampSecond.now() - 600))
+    val pendingPayments = pendingInvoices.map( invoice => IncomingPayment(invoice, randomBytes32(), PaymentType.Standard, invoice.createdAt.toTimestampMilli, IncomingPaymentStatus.Pending))
+    pendingInvoices.lazyZip(pendingPayments).foreach((invoice, payment) => db.addIncomingPayment(invoice, payment.paymentPreimage))
+
+    // create paid invoices
+    val receivedAt = TimestampMilli.now() + 1.milli
```

Comment:
I don't think you should rely on `TimestampMilli.now()` and `TimestampSecond.now()` (lines 48 and 55) to generate these invoices, you can use purely hard-coded values like you did for expired invoices, it will be less confusing.

Reply:
Good point; fixed in commit [600dfa](https://github.com/remyers/eclair/commit/600dfaa0e4941cfe9d2d19f9500dbe3cc7f88b97)

I only needed to set an invoice timestamp with  `TimestampMilli.now()` on line [118](https://github.com/remyers/eclair/commit/600dfaa0e4941cfe9d2d19f9500dbe3cc7f88b97#diff-5a95d11c0f8505b68f4a85b5a6dd3a1f9264880f632080db81c4c16ae2905400R118) in my second test to create an invoice that was expired within the 15 day look back period.